### DISCUSSION TITLE
Facing issue in Report module

### DISCUSSION THREAD

**Discussion by pallavii2809** [egovernments/Digit-Core#116] [Announcements]
Regarding the Reports for the PGR module, we are currently faced with a requirement to retrieve the beneficiary's name and mobile number (citizen) but we have encrypted data for all users, Within the pgr-service, we have a field called "accountId" that stores the UUID of the citizen, which is visible in the report. Our requirement is to populate this UUID when making an external service call for user search
So is there any way to have it done?

**Comment by subhashini-egov** [comment]
@shashwat-egov @talele08 can you please answer this?

**Reply by rishabhsuriwtc** [reply]
hi, @subhashini-egov do we have an update on it ?

**Comment by shashwat-egov** [comment]
Hi, if you are on the latest version of report, you can make the following changes to enable decryption of PII data like name by following these steps - 

1. Add the following decryptionPathId to report config - 
[Decryption Path Id Key](https://github.com/egovernments/configs/blob/qa/reports/config/rainmaker-obps-reports.yml#L4)
2. Ensure that name and uuid keys have been configured in the said report - 
[UUID key](https://github.com/egovernments/configs/blob/qa/reports/config/rainmaker-obps-reports.yml#L40)
[Name key](https://github.com/egovernments/configs/blob/qa/reports/config/rainmaker-obps-reports.yml#L35)
3. Add SecurityPolicy for the name field with the same name as the decryptionPathId provided in step 1 - 
[Security Policy](https://github.com/egovernments/egov-mdms-data/commit/d7fd41eefe28233002fce12a0dce7887e83e75ec)

If you are running older versions of report, you can refer to the commits and go about customizing report to cater to your use-case.

**Reply by rishabhsuriwtc** [reply]
Hey Shashwat, 

May i know which version of the report includes these changes ?

**Reply by shashwat-egov** [reply]
Does the build which you are running have this version - 1.3.5 ? If yes, it includes this functionality.

**Comment by pallavii2809** [comment]
Hii @shashwat-egov We are getting this error while running the Latest Report service, Can you please look into it 

![image](https://github.com/egovernments/Digit-Core/assets/109335293/484629ae-c6c7-44ca-a89f-fa166ffec5ab)

**Reply by rishabhsuriwtc** [reply]
@shashwat-egov this is happening when we are trying to upgrade to 1.3.5

**Comment by pallavii2809** [comment]
Hii @shashwat-egov do we have an update on the issue which we are facing in report while upgrading it to version - 1.3.5 ?

**Reply by shashwat-egov** [reply]
As mentioned in this comment - https://github.com/egovernments/Digit-Core/discussions/116#discussioncomment-6412524 , please ensure that you have the security policy json config in place.

**Reply by pallavii2809** [reply]
Thanks For your response, It's working fine now.
Could you please provide the necessary steps or any relevant documentation for enable decryption of PII data or this is all we need to do?

**Reply by shashwat-egov** [reply]
as far as report is concerned, you just need to do the steps mentioned in the aforementioned comment and it will handle the decryption for you.

**Reply by pallavii2809** [reply]
Hii @shashwat-egov  I have made all the required changes as you have mentioned after that I'm getting blank screen can you please guide on this ?

**Reply by pallavii2809** [reply]
Hii @shashwat-egov do we have an update on the issue?

**Reply by shashwat-egov** [reply]
can you check in your network tab so as to whether the API is giving you the proper response?

**Reply by pallavii2809** [reply]
Hii @shashwat-egov I'm getting this error

{
    "ResponseInfo": null,
    "Errors": [
        {
            "code": "ERROR_IN_RETRIEVING_REPORT_DATA",
            "message": "Invalid input: JSON Pointer expression must start with '/': \"uuid\"",
            "description": null,
            "params": null
        }
    ]
}

**Reply by shashwat-egov** [reply]
Hi Pallavi, there seems to be an issue with the jsonPath which you have provided in your security policy configuration. Please refer to this config - [Security Policy](https://github.com/egovernments/egov-mdms-data/blob/QA/data/pb/DataSecurity/SecurityPolicy.json#L693)

**Reply by pallavii2809** [reply]
Hii @shashwat-egov now I'm getting response 200 Ok
can you please check once 

![image](https://github.com/egovernments/Digit-Core/assets/109335293/d8168be7-d9fc-487e-bd08-1a829a229cd8)

**Reply by shashwat-egov** [reply]
As the API is returning 200 ok response, it seems to be fine.

**Reply by pallavii2809** [reply]
But I'm getting blank screen while searching the report 
![image](https://github.com/egovernments/Digit-Core/assets/109335293/c1cecca6-0bea-477f-925b-b81f3ac4c065)

**Reply by shashwat-egov** [reply]
This seems to be an issue with the frontend. Let me loop in @jagankumar-egov who is our frontend lead. He will help you with the loading issue.

**Reply by jagankumar-egov** [reply]
@pallavii2809
 can you share the report config for that specific report, Also the console error in that screen?

**Reply by pallavii2809** [reply]
Hii @shashwat-egov @jagankumar-egov  Now I'm getting reports with Decrypted data
Thanks for the help !

**Comment by shashwat-egov** [comment]
Closing this thread as the issue is resolved.
